spec/taskBehaviors.js

```diff
@@ -205,6 +206,74 @@ describe('Tasks', function() {
             expect(runValues).toEqual([1,3]);
         });
     });
+
+    describe('runEarly', function() {
+        it('Should run tasks early', function() {
+            var runValues = [];
+            var func = function(value) {
+                runValues.push(value);
+            };
+            ko.tasks.schedule(func.bind(null, 1));
+            expect(runValues).toEqual([]);
+
+            // Immediately runs any scheduled tasks
+            ko.tasks.runEarly();
+            expect(runValues).toEqual([1]);
+
+            // Skip calling jasmine.Clock.tick to show that the queue is clear
+        });
+
+        it('Should run tasks early during task processing', function() {
+            var runValues = [];
+            var func = function(value) {
+                runValues.push(value);
+            };
+
+            // Schedule two tasks; the first one schedules other tasks and calls runEarly
+            ko.tasks.schedule(function() {
+                ko.tasks.schedule(func.bind(null, 2));
+                expect(runValues).toEqual([]);
+
+                ko.tasks.runEarly();
+                expect(runValues).toEqual([1,2]);
+
+                // Schedule another task; it will be run after this one completes
+                ko.tasks.schedule(func.bind(null, 3));
+            });
+            ko.tasks.schedule(func.bind(null, 1));
+
+            jasmine.Clock.tick(1);
+            expect(runValues).toEqual([1,2,3]);
+        });
+
+        it('Should keep correct state if a task throws an exception', function() {
```

Comment:
I'm not sure that runEarly should throw if one of the handlers it runs throws. This means *everywhere* you use runEarly, you have to put all your code in try/catch, because you have no idea whether some completely unrelated task will throw. And people can use it as a way of detecting whether or not at least one scheduled handler does throw (without telling you *which* handler), which encourages excessive use of runEarly, undermining the benefit.

Could we change this so runEarly doesn't throw? Throwing doesn't seem to communicate anything useful, as you have no idea what threw (and I don't think we should add a way of reporting what threw, as it would lead to really weird use cases). Then all task exceptions are definitely going to land in the same place - the `onError` callback (or `window.onerror` if you catch at that level).